src/Client/PullPayment.php

```diff
@@ -0,0 +1,336 @@
+<?php
+
+declare(strict_types=1);
+
+namespace BTCPayServer\Client;
+
+class PullPayment extends AbstractClient
+{
+    /*
+    |--------------------------------------------------------------------------
+    | Management Pull Payment Methods.
+    |--------------------------------------------------------------------------
+    |
+    | For management to get, create and archive a pull payment.
+    | Approve, cancel or mark a payout as paid.
+    |
+    */
+
+    /**
+    * Get Store's Pull Payments
+    *
+    * @return string $id
+    * @return string $name
+    * @return string $currency
+    * @return int $amount
+    * @return int $period
+    * @return int $BOLT11Expiration
+    * @return bool $archived
+    * @return string $viewlink
+    */
+
+    public function getStorePullPayments(
+        string $storeId,
+        bool $includeArchived
+    ) {
+        $url = $this->getApiUrl() . 'stores/' .
+                    urlencode($storeId) . '/pull-payments';
+
+        $headers = $this->getRequestHeaders();
+        $method = 'GET';
+
+        $body = json_encode(
+            [
+                'includeArchived' => $includeArchived,
+            ],
+            JSON_THROW_ON_ERROR
+        );
+
+        $response = $this->getHttpClient()->request($method, $url, $headers, $body);
+
+        if ($response->getStatus() === 200) {
+            return json_decode($response->getBody(), true, 512, JSON_THROW_ON_ERROR);
```

Comment:
Please add a result class that extends the `AbstractResult` class so we have it consistent with the others. See `src/Result` for examples.